src/redecomp/RedecompMesh.cpp

```diff
@@ -54,20 +54,26 @@ RedecompMesh::RedecompMesh( const mfem::ParMesh& parent, double ghost_length, Pa
       SLIC_ERROR_ROOT( "Only 2D and 3D meshes are supported." );
   }
 
-  // preclude degenerate case where num elements/2 < num ranks
-  // factor of 2 on num elements are due to elements being paired for contact
-  auto n_parts = std::min( parent.GetNRanks(), ( static_cast<int>( parent.GetGlobalNE() ) + 1 ) / 2 );
-  p2r_elems_ = BuildP2RElementList( *partitioner, n_parts, ghost_length );
+  auto parent_n_els = static_cast<int>( parent.GetGlobalNE() );
+  if ( n_ranks == 0 ) {
```

Comment:
Can you clarify why this check is for n_ranks == 0? Is this just saying that n_ranks is initialized to 0 and then needs to be set, but if != 0 that means we have already set it appropriate according to this new criterion?

Reply:
See doxygen comment for what happens when n_ranks == 0.  I added some checks for preventing n_ranks < 0 and n_ranks > number of MPI ranks.